RxJavaTemplate/app/src/main/java/co/nimblehq/rxjava/storage/BaseSharedPreferencesImpl.kt

```diff
@@ -0,0 +1,39 @@
+package co.nimblehq.rxjava.storage
+
+import android.content.Context
+import android.content.SharedPreferences
+import android.preference.PreferenceManager
+import co.nimblehq.rxjava.extension.execute
+import co.nimblehq.rxjava.extension.put
+import javax.inject.Inject
+
+// TODO: For demo purpose only, replace with actual value that needs to be stored securely
+private const val PREF_TEST_ID = "PREF_TEST_ID"
+
+class BaseSharedPreferencesImpl @Inject constructor(applicationContext: Context) :
+    BaseSharedPreferences {
+
+    private var sharedPreferences: SharedPreferences
+
+    init {
+        sharedPreferences = defaultPreference(applicationContext)
+    }
+
+    // TODO: For demo purpose only, replace with actual value that needs to be stored securely
+    override var testId: String?
+        get() = sharedPreferences.getString(PREF_TEST_ID, null)
+        set(value) {
+            sharedPreferences.execute { it.put(PREF_TEST_ID to value) }
+        }
```

Comment:
If we have a dozen of values so we have to duplicate:

```
        get() = sharedPreferences.getXXX()
        set(value) {
            sharedPreferences.execute { it.put(XXX to value) }
        }
```

Could we have the function with simple `set(KEY_XXX, value)` and `get(KEY_XXX)` for common usage?